presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -217,17 +224,17 @@ public boolean isEnabled(Session session)
         public Result apply(FilterNode node, Captures captures, Context context)
         {
             JoinNode joinNode = captures.get(JOIN);
-            Expression filter = OriginalExpressionUtils.castToExpression(node.getPredicate());
-            List<FunctionCall> spatialFunctions = extractSupportedSpatialFunctions(filter);
-            for (FunctionCall spatialFunction : spatialFunctions) {
+            RowExpression filter = node.getPredicate();
+            List<CallExpression> spatialFunctions = extractSupportedSpatialFunctions(filter, functionManager);
```

Comment:
With this migration, the `extractSupportedSpatialFunctions` for `Expression` can be removed in class `SpatialJoinUtils`